### ISSUE TITLE
Shapes are not correctly positioned in the match collection

### ISSUE THREAD

**Issue by phet-steele** [phetsims/fraction-matcher#95]
The position of representations (circles, squares, etc...) in their gray box will be incorrect if they are included in the last match of a perfectly scored level. This was seen to happen on Witten (iPad2 9.2.1), diSessa (iPad3 9.2.1), and Tycho (iPad2 0.3 Beta 6).
I'm labeling this as a performance issue since the iPad Air 2 (Leibniz) does not show this problem. The Air blows the other iPads out of the water with its performance. There is a noticeable lag between pressing "OK" on the last match and when the reward cascade begins. During this lag is when the match animates towards the gray box and comes to rest incorrectly.

![20160315_122119](https://cloud.githubusercontent.com/assets/5863899/13789227/a9201858-eaa8-11e5-902a-d9a3746b1455.jpg)

Zoomed:
![20160315_122130](https://cloud.githubusercontent.com/assets/5863899/13789233/ade548ae-eaa8-11e5-9a12-d327df1cd06d.jpg)

Assigning to @ariel-phet to figure out who will deal with this. For phetsims/tasks/issues/541.

**Comment by phet-steele** [comment]
Troubleshooting information (do not edit):
Name: ‪Fraction Matcher‬
URL: http://www.colorado.edu/physics/phet/dev/html/fraction-matcher/1.1.0-rc.1/fraction-matcher_en.html
Version: 1.1.0-rc.1 2016-03-10 21:40:14 UTC
Features missing: fullscreen
User Agent: Mozilla/5.0 (iPad; CPU OS 9_3 like Mac OS X) AppleWebKit/601.1.46 (KHTML, like Gecko) Version/9.0 Mobile/13E5231a Safari/601.1
Language: en-us
Window: 1024x671
Pixel Ratio: 1/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 IMGSGX543-124.1)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.00)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 8 uniform: 128
Texture: size: 4096 imageUnits: 8 (vertex: 8, combined: 8)
Max viewport: 4096x4096
OES_texture_float: true
Dependencies JSON: {"assert":{"sha":"7d27130a","branch":"HEAD"},"axon":{"sha":"9457f36a","branch":"HEAD"},"babel":{"sha":"9c243fba","branch":"master"},"brand":{"sha":"ac9ccaa7","branch":"HEAD"},"chipper":{"sha":"b3a8a3c2","branch":"HEAD"},"dot":{"sha":"da78031b","branch":"HEAD"},"fraction-matcher":{"sha":"20c75954","branch":"HEAD"},"joist":{"sha":"28f51008","branch":"HEAD"},"kite":{"sha":"10ca8bf2","branch":"HEAD"},"phet-core":{"sha":"c48bf320","branch":"HEAD"},"phetcommon":{"sha":"380b1fb5","branch":"HEAD"},"scenery":{"sha":"b34b4fea","branch":"HEAD"},"scenery-phet":{"sha":"3d7d9459","branch":"HEAD"},"sherpa":{"sha":"bb68bbbc","branch":"HEAD"},"sun":{"sha":"7d41341a","branch":"HEAD"},"tandem":{"sha":"2441eec6","branch":"HEAD"},"vegas":{"sha":"5df94328","branch":"HEAD"},"vibe":{"sha":"c84b3c41","branch":"HEAD"}}

**Comment by ariel-phet** [comment]
@samreid seems worthy of  "quick" investigation, if 30 minutes yields nothing fruitful, please assign back to me.

**Comment by samreid** [comment]
My first step will be to try to reproduce this problem on my iPad3 and then on OS X Safari.

**Comment by samreid** [comment]
I am seeing this problem on iPad3, and can reliably reproduce it when the final match has a compound number in a graphic representation.

**Comment by samreid** [comment]
I cannot reproduce the problem on OS X Chrome or OS X Safari.

**Comment by samreid** [comment]
> @samreid seems worthy of "quick" investigation, if 30 minutes yields nothing fruitful, please assign back to me.

I'm 41 minutes in and can reliably reproduce the problem but only on iPad and only on certain levels, the suspicious code is:

```js
        [ 0, 1 ].forEach( function( i ) {
          var shape = thisNode.model.shapes[ thisNode.model.dropZone[ thisNode.model.gameModel.MAXIMUM_PAIRS * 2 + i ] ];
          var newPosition = thisNode.getShapeAnswerPosition( thisNode.model.answers.length );
          var initialScale = shape.view.matrix.scaleVector.x;
          var targetScale = initialScale / 2;
          var linearFunction = new LinearFunction( 0, 1, initialScale, targetScale );
          new TWEEN.Tween( shape.view ).easing( TWEEN.Easing.Cubic.InOut ).to( {
            scale: 0.5,
            centerX: newPosition.x,
            centerY: newPosition.y
          }, thisNode.model.gameModel.ANIMATION_TIME ).onUpdate( function( step ) {
            var scale = linearFunction( step );
            shape.view.setScaleMagnitude( scale, scale );
          } ).start();
          thisNode.model.answers.push( thisNode.model.dropZone[ thisNode.model.gameModel.MAXIMUM_PAIRS * 2 + i ] );
          thisNode.model.dropZone[ thisNode.model.gameModel.MAXIMUM_PAIRS * 2 + i ] = -1;
          shape.view.removeInputListener( shape.view.getInputListeners()[ 0 ] );
        } )
```

or

```js
      getShapeAnswerPosition: function( position ) {
        var targetRect = this.levelsContainer.answerRects[ Math.floor( position / 2 ) ];
        var diff = (position % 2 === 0) ? -targetRect.width / 4 : targetRect.width / 4;
        return new Vector2( targetRect.centerX + diff, targetRect.centerY );
      },
```

Or the way the tweens are used or managed.  I think it could be solved in 1-3 more hours.  @ariel-phet please advise, I'll wait to hear back from you before proceeding with RC.2.

**Comment by ariel-phet** [comment]
@samreid, yes, please solve before RC.2

I think in this case since the sim is made for an audience that often uses iPads (younger students) and the whole focus is equivalence, this bug is bothering me.

If trying to solve starts to get really messy, abandon and we will leave as is for now, move on with RC and prioritize for someone like Aadish to fix.

**Comment by samreid** [comment]
The problem does not appear on OS X Chrome with window.phet.joist.makeEverythingSlow()

**Comment by samreid** [comment]
I just reproduced this problem on Level 1.

**Comment by samreid** [comment]
Setting debugReward to true, I can reproduce this problem on the 1st match on Level 4 (didn't appear on level 1).

**Comment by samreid** [comment]
A fix for the tweened object seems like it solves this problem:

![image](https://cloud.githubusercontent.com/assets/679486/13921935/e505d25e-ef3f-11e5-8f12-1439bf24eca1.png)

**Comment by samreid** [comment]
A second test on iPad and another on OS X Chrome reveals this still looks good.  We can have @phet-steele verify it in the next RC.

**Comment by phet-steele** [comment]
Verified fixed in 1.1.0-rc.2.
